### DISCUSSION TITLE
How to generate apis types from existed crd definition yaml ?

### DISCUSSION THREAD

**Discussion by ArenaSu** [kubernetes-sigs/kubebuilder#3067] [Support] [answered]
How to generate apis types from existed crd definition yaml ? For example i have a crd definition yaml, and i want to generate golang apis types, is there a way to do it ?

The Scenario is below:
My project (generated by kubebuilder) import openfunction v0.7.0 which import keda(github.com/kedacore/keda/v2) v2.4.0 .Meanwhile my project import another inner project which import keda(github.com/kedacore/keda/v2) v2.8.1. Then it conflicts and error occurred when exec go mod tidy. I want to generate keda v2.4.0  api types from v2.4.0 crds and generate keda v2.8.1 api types from v2.8.1 crds. Then move the api types to different directories and import them inner.

And i know kubernetes-client Java have a action named "CRD Java Model Generate", it can generate apis types from crd definition yaml.

**Comment by camilamacedo86** [comment, marked as answer]
We need help to doc it, could you please raise an issue for that?
Also, please feel free to contribute with your help is very welcome.

Give a look at https://github.com/kubernetes-sigs/kubebuilder/blob/master/docs/using_an_external_type.md (old doc) we need a new doc with the info updated. Also, see that we need to change a tool to properly support it but you can do that by doing some manual changes in the scaffolds done: https://github.com/kubernetes-sigs/kubebuilder/pull/3055

**Reply by ArenaSu** [reply]
OK, i'll add an issue. As i am not clear on the internal implementation of kubebuilder, i'm sorry i can not help contributing it by myself in a short time.

**Reply by ArenaSu** [reply]
Issue link #3082

**Comment by camilamacedo86** [comment]
Closing since it is answered.
